Common/Source/Dialogs/dlgIGCProgress.cpp

```diff
@@ -0,0 +1,188 @@
+    
+  const auto oldFont = Surface.SelectObject(MapWindowBoldFont);
+
+  Surface.FillRect(&PrintAreaR, LKBrush_Petrol);
+
+  // Create text area
+
+  // we cannot use LKPen here because they are not still initialised for startup menu. no problem
+  LKPen hP(PEN_SOLID,NIBLSCALE(1),RGB_GREEN);
+  auto ohP = Surface.SelectObject(hP);
+  const auto ohB = Surface.SelectObject(LKBrush_Petrol);
+  Surface.Rectangle(PrintAreaR.left,PrintAreaR.top,PrintAreaR.right,PrintAreaR.bottom);
+  Surface.SelectObject(ohP);
+  hP.Release();
+
+  hP.Create(PEN_SOLID,NIBLSCALE(1),RGB_BLACK);
+  ohP = Surface.SelectObject(hP);
+  Surface.SelectObject(LK_HOLLOW_BRUSH);
+  InflateRect(&PrintAreaR, -NIBLSCALE(2), -NIBLSCALE(2));
+  Surface.Rectangle(PrintAreaR.left,PrintAreaR.top,PrintAreaR.right,PrintAreaR.bottom);
+
+  Surface.SetTextColor(RGB_WHITE);
+  Surface.SetBackgroundTransparent();
+
+  InflateRect(&PrintAreaR, -NIBLSCALE(2), -NIBLSCALE(2));
+  
+  const TCHAR* text = Sender->GetCaption();
+  Surface.DrawText(text, &PrintAreaR, DT_VCENTER|DT_SINGLELINE);
+
+  Surface.SelectObject(ohB);
+  Surface.SelectObject(ohP);
+  Surface.SelectObject(oldFont);
+
+}
+
+CallBackTableEntry_t IGCProgressCallBackTable[] = {
+    OnPaintCallbackEntry(OnIGCSplashPaint),
+    OnPaintCallbackEntry(OnIGCProgressPaint),
+    ClickNotifyCallbackEntry(OnIGCAbortClicked),
+    EndCallBackEntry()
+};
+
+class dlgIGCProgress final  {
+public:
+    dlgIGCProgress();
+    ~dlgIGCProgress();
+    
+    dlgIGCProgress( const dlgIGCProgress& ) = delete;
+    dlgIGCProgress& operator=( const dlgIGCProgress& ) = delete;
+
+};
+
+
+
+
+
+dlgIGCProgress::dlgIGCProgress() {};
+void dlgIGCProgressShowModal(void){
+    
+	bClose = false;
+
+    _WndForm = dlgLoadFromXML(IGCProgressCallBackTable, ScreenLandscape ? IDR_XML_IGC_PROGRESS_P : IDR_XML_IGC_PROGRESS_L);
+
+    LKASSERT(_WndForm);
+    if(_WndForm) {
+        WindowControl* wSplash = _WndForm->FindByName(TEXT("frmIGCSplash")); 
+        if(wSplash) {
+            wSplash->SetWidth(_WndForm->GetWidth());
+            wSplash->SetHeight(_WndForm->GetHeight());
+        }
+        WindowControl* wText = _WndForm->FindByName(TEXT("frmIGCText")); 
+        if(wText) {
+            wText->SetWidth(_WndForm->GetWidth());
+            wText->SetTop(_WndForm->GetHeight() - wText->GetHeight());
+        }
+        _WndForm->Show();
+        _WndForm->Redraw();
+        _WndForm->SetTimerNotify(200, OnIGCProgressTimer); // check for end of download every 200ms
+    }
+    _WndForm->SetTimerNotify(200, OnIGCProgressTimer); // check for end of download every 200ms
+    _WndForm->ShowModal();
+
+    delete _WndForm;
+    _WndForm = NULL;
+
+}
+
+dlgIGCProgress::~dlgIGCProgress() {
+    _WndForm->SetTimerNotify(0,NULL); // delete Timer
+    delete _WndForm; 
+}
+
+
+
+
+void CloseIGCProgressDialog() {
+    bClose = true;
+
+}
+
+void CreateIGCProgressDialog(const TCHAR* text) {
+   dlgIGCProgressShowModal();
+   IGCProgressDialogText(text);
```

Comment:
useless 
dlgIGCProgressShowModal() are blocking and return only return after set "_WndForm" to null, so IGCProgressDialogText do nothing and text parameter are never used.

Reply:
It is definitly working on LINUX, PC and KOBO, but I don't argue so I removed it.
done.

Reply:
it work for sure, a function doing nothing can't failed ....

Reply:
I mean it showed the text!

Reply:
- last line of "dlgIGCProgressShowModal()" is  "_WndForm = NULL;"
- IGCProgressDialogText() do nothing  if "_WndForm" is false 

so if "text" is showed, it's by side effect elsewhere ...